### PR TITLE
Encapsulate RPC Result

### PR DESCRIPTION
**Pull request by nirav1911**
#### Problem
What is being fixed?  Examples:
* Added introspection support for RPC Client

#### Change overview
What's in this PR

Added introspection support for RPC Client

#### Testing
How was this tested? (at least one bullet point required)
*Tested with test_client/rpc_client.py

### CODE REVIEW

File: src/controller/CHIPDeviceController.cpp [hunk 1]
```diff
@@ -503,6 +503,16 @@ CHIP_ERROR DeviceController::ServiceEventSignal()
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR DeviceController::GetFabricId(uint64_t & fabricId)
```
Comment (nirav1911) [inline src/controller/CHIPDeviceController.cpp:506]: This file is not changed but I have worked on csg/test_event_3.1_fabric_id as base so all these changes are showing up in this PR.

File: src/controller/CHIPDeviceController.h [hunk 2]
```diff
@@ -253,6 +253,15 @@ class DLL_EXPORT DeviceController : public Messaging::ExchangeDelegate,
      */
     CHIP_ERROR ServiceEventSignal();
 
+    /**
```
Comment (nirav1911) [inline src/controller/CHIPDeviceController.h:256]: This file is not changed but I have worked on csg/test_event_3.1_fabric_id as base so all these changes are showing up in this PR.

File: src/controller/python/ChipDeviceController-ScriptBinding.cpp [hunk 3]
```diff
@@ -88,6 +88,7 @@ CHIP_ERROR pychip_DeviceController_DeleteDeviceController(chip::Controller::Devi
 CHIP_ERROR
 pychip_DeviceController_GetAddressAndPort(chip::Controller::DeviceCommissioner * devCtrl, chip::NodeId nodeId, char * outAddress,
                                           uint64_t maxAddressLen, uint16_t * outPort);
+CHIP_ERROR pychip_DeviceController_GetFabricId(chip::Controller::DeviceCommissioner * devCtrl, uint64_t * outFabricId);
```
Comment (nirav1911) [inline src/controller/python/ChipDeviceController-ScriptBinding.cpp:91]: This file is not changed but I have worked on csg/test_event_3.1_fabric_id as base so all these changes are showing up in this PR.

File: src/controller/python/chip-device-ctrl.py [hunk 4]
```diff
@@ -872,6 +896,7 @@ def start_rpc_server():
         server.register_function(qr_code_parse)
         server.register_function(get_pase_data)
         server.register_multicall_functions()
+        server.register_introspection_functions()
```
Comment (nirav1911) [inline src/controller/python/chip-device-ctrl.py:899]: This is the only change in `chip-device-ctrl.py` file as part of this PR.

File: src/controller/python/test_client/rpc_client.py [hunk 5]
```diff
@@ -64,6 +64,16 @@ async def create_container(docker_image_tag: str,  port: int) -> Container:
 
     return container
 
+def call_rpc(device_server,  method_name, *params):
```
Comment (nirav1911) [inline src/controller/python/test_client/rpc_client.py:67]: Added this method to test call to RPC Server.

**Review by mikaelhm — CHANGES_REQUESTED** [review]
This looks good, but I think we need to make clean branch of `csg/test_event_3.1` and cherry-pick the commit, so we don't merge the changes from `csg/test_event_3.1_fabric_id` before they're ready.

**Comment by nirav1911** [comment]
> This looks good, but I think we need to make clean branch of `csg/test_event_3.1` and cherry-pick the commit, so we don't merge the changes from `csg/test_event_3.1_fabric_id` before they're ready.

Created a new PR: https://github.com/chip-csg/connectedhomeip/pull/39

**Pull request closed without merging**